File: code_wars/kyu_6/004-Unique_In_Order.js
//###### Solution 1
var uniqueInOrder = function (iterable) {
  let unique_seq = [];

  for (let i = 0; i < iterable.length; i++) {
    if (i === 0 || iterable[i] !== iterable[i - 1])
      unique_seq.push(iterable[i]);
  }

  return unique_seq;
};

//###### Solution 2
var uniqueInOrder = function (iterable) {
  if (typeof iterable === "string" || iterable instanceof String)
    iterable = iterable.split("");
  return iterable.filter((ele, i) => i === 0 || ele !== iterable[i - 1]);
};

//###### Link: Video 008
//###### Problem Link: https://www.codewars.com/kata/54e6533c92449cc251001667/train/javascript

File: code_wars/kyu_6/004-Unique_In_Order.py
###### Solution 1
def unique_in_order(sequence):
    unique_seq = []

    for prev, item in enumerate(sequence):
        if prev == 0 or sequence[prev - 1] != item:
            unique_seq.append(item)

    return unique_seq


###### Solution 2
def unique_in_order(sequence):
    return [
        item
        for prev, item in enumerate(sequence)
        if prev == 0 or sequence[prev - 1] != item
    ]


###### 08
###### Problem Link: https://www.codewars.com/kata/54e6533c92449cc251001667/train/python
